doc/overview/security.md

```diff
@@ -54,8 +56,157 @@ permissions for specific actions.
+  * Allow the UNIX user named daos_user to have read-write access.
+* `A:G:project_users@:tc`
+  * Allow anyone in the UNIX group project_users to access a pool's contents and
+    create containers.
+* `A::OWNER@:rwdtTaAo`
+  * Allow the UNIX user who owns the container to have full control.
+* `A:G:GROUP@:rwdtT`
+  * Allow the UNIX group that owns the container to read and write data, delete
+    the container, and manipulate container properties.
+* `A::EVERYONE@:r`
+  * Allow any user not covered by other rules to have read-only access.
+* `A::daos_user@:`
+  * Deny the UNIX user named daos_user any access to the resource.
+
+#### Enforcement
+
+Access Control Entries (ACEs) will be enforced in the following order:
+
+* Owner-User
+* Named users
+* Owner-Group and named groups
+* Everyone
+
+In general, enforcement will be based on the first match, ignoring
+lower-priority entries.
+
+If the user is the owner of the resource, and there is an OWNER@ entry, they
+will receive the owner permissions only. They will not receive any of the
+permissions in the named user/group entries, even if they would match those
+other entries.
+
+If the user isn't the owner, or there is no OWNER@ entry, but there is an ACE
+for their user identity, they will receive the permissions for their user
+identity only. They will not receive the permissions for any of their
+groups, even if those group entries have broader permissions than the user entry
+does. The user is expected to match at most one user entry.
+
+If no matching user entry is found, but entries match one or more of the user's
+groups, enforcement will be based on the union of the permissions of all
+matching groups, including the owner-group.
+
+If no matching groups are found, the "Everyone" entry's permissions will be
+used, if it exists.
+
+By default, if a user matches no ACEs in the list, access will be denied.
+
+#### ACL File
+
+Tools that accept an ACL file expect it to be a simple text file with one ACE
+on each line. A line may be marked as a comment by adding a `#` as the first
+non-whitespace character on the line.
+
+For example:
+
+```
+# ACL for my container
+# Owner can't touch data - just do admin-type things
+A::OWNER@:dtTaAo
+# My project's users can generate and access data
+A:G:my_great_project@:rw
+# Bob can use the data to generate a report
+A::bob@:r
+```
+
+That the permission bits and the ACEs themselves don't need to be in any
+specific order. However the order may be different when the resulting ACL is
+parsed and displayed by DAOS.
+
+#### Limitations
+
+The maximum size of the ACE list in a DAOS ACL internal data structure is 64KiB.
+
+To calculate the internal data size of an ACL, use the following formula for
+each ACE:
+
+* The base size of an ACE is 256 bytes.
+* If the ACE principal is *not* one of the special principals:
+  * Add the length of the principal string + 1.
+  * If that value is not 64-byte aligned, round up to the nearest 64 byte 
```

Comment:
(style)  trailing whitespace

Reply:
Should probably resolve this whitespace issue